evennia/typeclasses/tags.py

```diff
@@ -319,6 +319,37 @@ def add(self, tag=None, category=None, data=None):
             getattr(self.obj, self._m2m_fieldname).add(tagobj)
             self._setcache(tagstr, category, tagobj)
 
+    def has(self, tag=None, category=None, return_list=False):
+        """
+        Checks if the given Tag (or list of Tags) exists on the object.
+
+        Args:
+            tag (str or iterable): The Tag key or tags to check for.
+                If `None`, search by category.
+            category (str or None): Limit the check to Tags with this
+                category (note, that `None` is the default category).
+
+        Returns:
+            has_tag (bool or list): If the Tag exists on this object or not.
+             If `tag` was given as an iterable then the return is a list of booleans.
```

Comment:
Good documentation string! :+1: 

Only thing that is missing is a `Raises:` section since this raises `ValueError` on invalid input. No big deal, I'll add that as I merge.